src/client/api/kernelApi.ts

```diff
@@ -35,10 +35,12 @@ export class JupyterKernelServiceFactory {
         @inject(IKernelProvider) private readonly kernelProvider: IKernelProvider,
         @inject(IDisposableRegistry) private readonly disposables: IDisposableRegistry,
         @inject(INotebookControllerManager) private readonly notebookControllerManager: INotebookControllerManager,
-        @inject(ApiAccessService) private readonly apiAccess: ApiAccessService
+        @inject(ApiAccessService) private readonly apiAccess: ApiAccessService,
+        @inject(IExtensions) private readonly extensions: IExtensions
     ) {}
     public async getService() {
-        const accessInfo = await this.chainedApiAccess.chainFinally(() => this.apiAccess.getAccessInformation());
+        const info = await this.extensions.determineExtensionFromCallStack();
```

Comment:
This has to happen before the 'chainedApiAccess' promise or the callstack doesn't return anything.